app/src/main/java/org/p2p/wallet/settings/ui/devices/DeviceCellMapper.kt

```diff
@@ -0,0 +1,137 @@
+package org.p2p.wallet.settings.ui.devices
+
+import androidx.annotation.StringRes
+import kotlinx.coroutines.withContext
+import org.p2p.core.common.DrawableContainer
+import org.p2p.core.common.TextContainer
+import org.p2p.uikit.components.finance_block.MainCellModel
+import org.p2p.uikit.components.icon_wrapper.IconWrapperCellModel
+import org.p2p.uikit.components.left_side.LeftSideCellModel
+import org.p2p.uikit.components.right_side.RightSideCellModel
+import org.p2p.uikit.model.AnyCellItem
+import org.p2p.uikit.organisms.sectionheader.SectionHeaderCellModel
+import org.p2p.uikit.utils.dip
+import org.p2p.uikit.utils.drawable.DrawableCellModel
+import org.p2p.uikit.utils.drawable.shape.shapeRoundedAll
+import org.p2p.uikit.utils.drawable.shapeDrawable
+import org.p2p.uikit.utils.image.ImageViewCellModel
+import org.p2p.uikit.utils.text.TextViewBackgroundModel
+import org.p2p.uikit.utils.text.TextViewCellModel
+import org.p2p.uikit.utils.text.TextViewSize
+import org.p2p.uikit.utils.text.badgePadding
+import org.p2p.uikit.utils.text.badgeRounded
+import org.p2p.wallet.R
+import org.p2p.wallet.infrastructure.dispatchers.CoroutineDispatchers
+import org.p2p.wallet.utils.toPx
+
+class DeviceCellMapper(
+    private val dispatchers: CoroutineDispatchers
+) {
+
+    suspend fun toCellModels(
+        currentDeviceName: String,
+        oldDeviceName: String
+    ): List<AnyCellItem> = withContext(dispatchers.io) {
+
+        val cells = mutableListOf<AnyCellItem>()
+
+        cells += createHeaderCell()
+
+        cells += createSectionCell(R.string.devices_this_device)
+
+        cells += createDeviceCell(currentDeviceName, buttonRes = R.string.devices_set_up)
+
+        cells += createSectionCell(R.string.devices_authorization_device)
+
+        cells += createDeviceCell(oldDeviceName, R.string.devices_make_sure_your_device)
+
+        cells += createFooterCell()
+
+        return@withContext cells
```

Comment:
no need for labeled @ return
```suggestion
        cells
```

or I can suggest doing buildList and calling this += createHeaderCell and etc

Reply:
Adding the label is okay too. 
It helps to understand that the current function is wrapped into asynchronous thread